### PR TITLE
Fix weak comparison between `'0'` and `false`

### PR DESCRIPTION
**Pull request by mvorisek**
fixes https://github.com/sebastianbergmann/phpunit/issues/4967

### CODE REVIEW

**Comment by mvorisek** [comment]
@sebastianbergmann can you please merge this fix into 3.x - 5.x to fix phpunit 8.5 and 9.5?

File: src/ScalarComparator.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ public function assertEquals($expected, $actual, $delta = 0.0, $canonicalize = f
 
         // always compare as strings to avoid strange behaviour
```
Comment (alexislefebvre) [inline src/ScalarComparator.php:52]: This behaviour exist since 9 years, changing it may be a BC break.

Comment (mvorisek) [reply src/ScalarComparator.php:52]: that is true, but `0` equals `false`, so `'0'` must too, it is also the php behaviour of weak comparison

**Comment by sebastianbergmann** [comment]
Thank you.

Versions 3.0.4 and 4.0.7 have been tagged, https://github.com/sebastianbergmann/phpunit/pull/4972 can be updated.

**Pull request merged**